app/models/campagne.rb

```diff
@@ -10,7 +10,8 @@ class Campagne < ApplicationRecord
 
   validates :parcours_type, presence: true, on: :create
   validates :libelle, presence: true
-  validates :code, presence: true, uniqueness: { case_sensitive: false }
+  validates :code, presence: true, uniqueness: { case_sensitive: false },
+                   format: { with: /\A[a-zA-Z0-9]+\z/, message: 'Lettres et chiffres uniquement' }
```

Comment:
Le code est forcement avec des majuscules, j'aurais donc tendance à empecher de mettre des minuscules comme ceci : 

```suggestion
                   format: { with: /\A[A-Z0-9]+\z/, message: 'Lettres et chiffres uniquement' }
```